src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ExceptionInfoLookupTableNode.cs

```diff
@@ -0,0 +1,153 @@
+            sb.Append(Offset.ToString());
+        }
+
+        internal void LayoutMethodsWithEHInfo()
+        {
+            if (_methodNodes != null)
+            {
+                // Already initialized
+                return;
+            }
+
+            _methodNodes = new List<MethodWithGCInfo>();
+            _ehInfoOffsets = new List<int>();
+
+            foreach (MethodDesc method in _nodeFactory.MetadataManager.GetCompiledMethods())
+            {
+                MethodWithGCInfo methodCodeNode = _nodeFactory.MethodEntrypoint(method) as MethodWithGCInfo;
+                if (methodCodeNode == null)
+                {
+                    methodCodeNode = ((ExternalMethodImport)_nodeFactory.MethodEntrypoint(method))?.MethodCodeNode;
+                    if (methodCodeNode == null)
+                        continue;
+                }
+
+                ObjectData ehInfo = methodCodeNode.EHInfo;
+                if (ehInfo != null && ehInfo.Data.Length != 0)
+                {
+                    _methodNodes.Add(methodCodeNode);
+                    _ehInfoOffsets.Add(_ehInfoNode.AddEHInfo(ehInfo.Data));
+                }
+            }
+        }
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            if (relocsOnly)
+            {
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolDefinitionNode[] { this });
+            }
+
+            if (_methodNodes == null)
+                LayoutMethodsWithEHInfo();
+
+            ObjectDataBuilder exceptionInfoLookupBuilder = new ObjectDataBuilder(factory, relocsOnly);
+            exceptionInfoLookupBuilder.RequireInitialAlignment(8);
+
+            // Add the symbol representing this object node
+            exceptionInfoLookupBuilder.AddSymbol(this);
+
+            // First, emit the actual EH records in sequence and store map from methods to the EH record symbols
+            for (int index = 0; index < _methodNodes.Count; index++)
+            {
+                exceptionInfoLookupBuilder.EmitReloc(_methodNodes[index], RelocType.IMAGE_REL_BASED_ADDR32NB);
+                exceptionInfoLookupBuilder.EmitReloc(_ehInfoNode, RelocType.IMAGE_REL_BASED_ADDR32NB, _ehInfoOffsets[index]);
+            }
+
+            // Sentinel record - method RVA = -1, EH info offset = total EH info size
+            exceptionInfoLookupBuilder.EmitUInt(~0u);
+            exceptionInfoLookupBuilder.EmitUInt((uint)_ehInfoNode.Count);
+
+            return exceptionInfoLookupBuilder.ToObjectData();
+        }
+
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory)
+        {
+            return new DependencyNodeCore<NodeFactory>.DependencyList(new DependencyListEntry[] { new DependencyListEntry(_ehInfoNode, "EH info array") });
+        }
+
+        public override int ClassCode => -855231428;
```

Comment:
Looks like a duplicate random number from RuntimeFunctionsTableNode.cs here

Reply:
Thank for spotting, fixed!